traffic_monitor/health/cache.go

```diff
@@ -53,14 +53,44 @@ func GetVitals(newResult *cache.Result, prevResult *cache.Result, mc *tc.Traffic
 		return
 	}
 
+	if mc == nil {
+		log.Errorf("mc TrafficMonitorConfigMap cannot be nil")
+		return
+	}
+
 	if newResult.InterfaceVitals == nil {
 		newResult.InterfaceVitals = map[string]cache.Vitals{}
 	}
 
 	// proc.loadavg -- we're using the 1 minute average (!?)
 	newResult.Vitals.LoadAvg = newResult.Statistics.Loadavg.One
 
-	for ifaceName, iface := range newResult.Interfaces() {
+	if mc.TrafficServer[newResult.ID].Interfaces == nil {
```

Comment:
Could you add some checks here to see that the length of `mc.TrafficServer` is not 0, and also to make sure that `mc.TrafficServer` has the key `newResult.ID`?

Reply:
This won't necessarily panic or cause any issues if the length is 0 or the cache isn't found. However, this is actually silently failing for those reasons. I will put checks in so that we can at least log the information to uncover the issues in the future should the cache map be empty or if the cache we're attempting to monitor is missing from said map.

Reply:
Done